### ISSUE TITLE
Why does TypeScript allow the const modifier on class member variable declarations?

### ISSUE THREAD

**Issue by bcherny** [Microsoft/TypeScript#25123]
But only when the class is anonymous. Is this expected behavior? I didn't see this documented in the [spec](https://github.com/Microsoft/TypeScript/blob/master/doc/spec.md#82-members) or the [handbook](https://www.typescriptlang.org/docs/handbook/classes.html).

**TypeScript Version:**  3.0.0-dev.201xxxxx

**Search Terms:** property, variable, let, var, lenient

**Code**

```ts
class _ {
  var a = 4 // ERROR
}
class _ {
  let a = 4 // ERROR
}
class _ {
  const a = 4 // ERROR
}
let _ = class {
  var a = 4 // ERROR
}
let _ = class {
  let a = 4 // ERROR
}
let _ = class {
  const a = 4 // OK???
}
```

**Expected behavior:**

The last case should produce an error.

**Actual behavior:**

The last case is allowed, and compiles to an instance variable as you might expect.

**Playground Link:** http://www.typescriptlang.org/play/#src=let%20_%20%3D%20class%20%7B%0D%0A%20%20const%20a%20%3D%204%20%2F%2F%20OK%3F%3F%3F%0D%0A%7D

**Related Issues:** -
